bee-message/src/address/bls.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::error::ValidationError;
+
+use bee_packable::Packable;
+
+use alloc::borrow::ToOwned;
+use core::{convert::TryInto, str::FromStr};
+
+/// The number of bytes in a BLS address.
+pub const BLS_ADDRESS_LENGTH: usize = 49;
+
+/// A BLS address.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Packable)]
+#[cfg_attr(feature = "enable-serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct BlsAddress(
+    #[cfg_attr(feature = "enable-serde", serde(with = "serde_big_array::BigArray"))] [u8; BLS_ADDRESS_LENGTH],
+);
+
+#[allow(clippy::len_without_is_empty)]
+impl BlsAddress {
+    /// The address kind of a BLS address.
+    pub const KIND: u8 = 1;
+
+    /// Creates a new BLS address.
+    pub fn new(address: [u8; BLS_ADDRESS_LENGTH]) -> Self {
+        address.into()
+    }
+
+    /// Returns the length of a BLS address.
+    pub const fn len(&self) -> usize {
+        BLS_ADDRESS_LENGTH
+    }
+
+    // TODO verification
```

Comment:
will this TODO be part of a second PR or was it forgotten?

Reply:
That should come later on, there are no BLS PK/signature types in crypto.rs right now, but when there are we can do this.

Reply:
ah understand, thanks for the info!